Backbase/variants
# refactor: remove the keyword and syntax previously necessary to work with environment variables in favour of an additional optional property 'env'

### What does this PR do
- Remove the syntax necessary to work with environment variables in custom properties;
- Add optional property `env` to `CustomProperty` type, whose default value is set to `false`;
- Internally, decode optional Bool to non-optional `isEnvironmentVariable`.

### How can it be tested
- Adhere to the [latest changes](https://github.com/Backbase/variants/commit/2bad20dc1ab9ac6275daf6db6e1ba475931063bb#diff-88e254e2c28df11e815768543a21790e7d60b64b331abcf1d324ef6f768e4043) in variants.yml
- Include a custom property whose value reads from environment variable:
```
 - name: sample_property
    value: NAME_OF_ENV_VAR
    env: true
    destination: project
```
- Switch to this respective variant variants switch --variant A_VARIANT
- **IF iOS** `Variants.swift` should contain `struct Secrets` with your encrypted static var.
- **IF Android** `gradleScripts/variants.gradle` should contain:
```
System.getenv('NAME_OF_ENV_VAR')
```

### Task
resolves #121 

### Checklist:
